pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -904,6 +894,20 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env *model.E
 	return append(tcpListeners, httpListeners...)
 }
 
+// Base used to remap privileged ports. 15000 is the range used by Istio special ports.
+const privPortRemap = 15200
+
+// maybeMapPort may remap privileged ports. Proxy doesn't run as root, so if bind=true ports <1024 will
+// not be accepted, envoy will reject the full config.
```

Comment:
this is not always true. I could choose to launch envoy without the uuid parameter if I am not using IPtables. I could be running this envoy on a VM or several other platforms such as CF/consul where this implicit remapping will break. Not to mention, it implicitly overrides the user specified intent (bind listener on port X).

If you locate the instance by the port name, then you can skip this entire logic and make the user define a non privileged inbound listener port for the workload.

Reply:
We could use a meta to indicate envoy is running as root.
But if it is not - and 99% of users should not run envoy as root - it would fail to start if ports < 1024 are used. Between 'ports are remapped' and 'nothing works' - I would rather default to the safe option, in particular given that this 'iptable-less' mode will be gradually adopted.

We can add a flag/env to control default and configure the base. 

K8S also remaps ports for NodePort.

Reply:
I think we are agreeing on a meta from envoy to customize the base, will be 0 if running as root.